entity-framework/core/what-is-new/ef-core-3.0/breaking-changes.md

```diff
@@ -1407,3 +1407,33 @@ These methods are used by EF to determine if a database is created but empty. Th
 **Mitigations**
 
 Change the accessibility of any overrides.
+
+## Microsoft.EntityFrameworkCore.Design is now a DevelopmentDependency package
+
+[Tracking Issue #11506](https://github.com/aspnet/EntityFrameworkCore/issues/11506)
+
+This change is introduced in EF Core 3.0-preview 4.
+
+**Old behavior**
+
+Before EF Core 3.0, Microsoft.EntityFrameworkCore.Design was a regular NuGet package whose assembly could be referenced by projects that depended on it.
+
+**New behavior**
+
+Starting with EF Core 3.0, it is a DevelopmentDependency package. Which means that the dependency won't flow transitively into other projects, and that you can no longer, by default, reference its assembly.
+
+**Why**
+
+This package is only intended to be used at design time. Deployed applications shouldn't reference it. Making the package a DevelopmentDependency reinforces this recommendation.
+
+**Mitigations**
+
+If you need to reference this package to override EF Core's design-time behavior, you can update update PackageReference item metadata in your project. If the package is being referenced transitively via Microsoft.EntityFrameworkCore.Tools, you will need to add an explicit PackageReference to the package to change its metadata.
```

Comment:
What types do you need access to in order to override EF Core's design-time behavior. Would it be enough to move some interfaces to the abstractions package? I remember vaguely that we already discussion this, but can't remember what the answer was.

Reply:
IPluralizer (if you’re not using something like my extension package), C# generators if you’re doing something pretty advanced... Nothing else I can think of

Reply:
That sounds like it would be reasonable to move. IDesignTimeDbContextFactory is there too?

Reply:
I created https://github.com/aspnet/EntityFrameworkCore/issues/16248. We can discuss what other things are required often as application dependencies in triage.

Reply:
IDesignTimeDbContextFactory is in Core